resources/windows/wix/properties.wxs.erb

```diff
@@ -0,0 +1,127 @@
+      Id="PUPPET_AGENT_ACCOUNT_USER"
+      Value="LocalSystem"/>
+    <Property
+      Id="PUPPET_AGENT_ACCOUNT_PASSWORD"
+      Hidden="yes"/>
+
+    <!-- To use WixUI_InstallDir, you must set a property named
+       WIXUI_INSTALLDIR with a value of the ID of the directory you want the
+       user to be able to specify the location of. -->
+    <Property
+      Id="WIXUI_INSTALLDIR"
+      Value="INSTALLDIR" />
+    <!-- This text is placed as a literal string in the GUI -->
+    <Property
+      Id="WIXUI_EXITDIALOGOPTIONALTEXT"
+      Value="Manage your first resources on this node, explore the Puppet community and get support using the shortcuts in the Documentation folder of your Start Menu." />
+
+    <!--
+      Remembered Property Pattern
+      http://robmensching.com/blog/posts/2010/5/2/The-WiX-toolsets-Remember-Property-pattern
+      -->
+    <!-- Registry settings, remembered install dir is dependent on x86/64.
+         Everything else stays with x86 so both installer types can see it
+      -->
+    <!-- Note the static default value of "puppet".  This will be overriden by
+         the command line. -->
+    <Property Id="PUPPET_MASTER_SERVER">
+      <!-- Recall the property in repair, upgrade, and uninstall scenarios -->
+      <RegistrySearch
+        Id="RecallPuppetMasterServer"
+        Root="HKLM"
+        Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:common_product_id] %>"
+        Name="RememberedPuppetMasterServer"
+        Type="raw"
+        Win64="no" />
+    </Property>
+    <Property Id="PUPPET_AGENT_ENVIRONMENT">
+      <!-- Recall the property in repair, upgrade, and uninstall scenarios -->
+      <RegistrySearch
+        Id="RecallPuppetAgentEnvironment"
+        Root="HKLM"
+        Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:common_product_id] %>"
+        Name="RememberedPuppetAgentEnvironment"
+        Type="raw"
+        Win64="no" />
+    </Property>
+
+    <Property Id="PUPPET_AGENT_CERTNAME">
+      <!-- Recall the property in repair, upgrade, and uninstall scenarios -->
+      <RegistrySearch
+        Id="RecallPuppetAgentCertname"
+        Root="HKLM"
+        Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:common_product_id] %>"
+        Name="RememberedPuppetAgentCertname"
+        Type="raw"
+        Win64="no" />
+    </Property>
+
+    <Property Id="PUPPET_CA_SERVER">
+      <RegistrySearch
+        Id="RecallPuppetCAServer"
+        Root="HKLM"
+        Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:common_product_id] %>"
+        Name="RememberedPuppetCaServer"
+        Type="raw"
+        Win64="no" />
+    </Property>
+
+    <Property Id="PUPPET_AGENT_STARTUP_MODE">
+      <RegistrySearch
+        Id="RecallPuppetAgentStartupMode"
+        Root="HKLM"
+        Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:common_product_id] %>"
+        Name="RememberedPuppetAgentStartupMode"
+        Type="raw"
+        Win64="no" />
```

Comment:
@McdonaldSeanp Sean - not sure about that, but will note this when I'm doing the testing with the UI - and will check with @ferventcoder 
I suspect it has to do with where we decide to store the registry keys (have you looked up windows system redirection - if not, its a wonderful topic).